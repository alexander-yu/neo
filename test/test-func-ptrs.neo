var func<(string):void> f;

int foo(int i) {
    return 2 * i;
}

void print_name(string s) {
    print("My name is ");
    print(s);
}

void main() {
    print(print);
    print("\n");
    print(foo);
    print("\n");

    create array<func<(int):int>> a = {| foo, foo |};
    create array<array<func<(string):void>>> b =
        {| {| print, print_name |}, {| print_name |} |};

    print(a);
    print("\n");
    print(b[0]);
    print("\n");
    b[0][0]("Hello!\n");

    var func<(int):int> g = a[0];
    print(g(3));
    print("\n");

    f = b[1][0];
    f("Alex");
}
