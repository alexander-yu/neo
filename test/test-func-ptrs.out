built-in function print
function at 0x4008f0
{|function at 0x4008f0, function at 0x4008f0|}
{|function at 0x400c0c, function at 0x400900|}
Hello!
6
My name is Alex
